e2e-test/e2e.py

```diff
@@ -96,6 +99,71 @@ def test_post_resource_increase_count(
         "%s %ss returned for %s", value_from_proxy, resource_search_pair[0], patient_id,
     )
 
+@patch.dict(os.environ, {"AUDIT_EVENT_LOGGING_ENABLED_ENV": "true"})
+def test_post_resource_with_logging_enabled_creates_audit_event(
+    file_name: str,
+    hapi: clients.HapiClient,
+    fhir_proxy: clients.FhirProxyClient,
+    auth: clients.AuthClient,
+) -> None:
+    """Test to add a resource to the backend via the Proxy and verify AuditEvent creation."""
+    token = auth.get_auth_token()
+ 
+    initial_audit_event_count = hapi.get_audit_event_count()
+    logging.info("Initial AuditEvent count: %d", initial_audit_event_count)
+
+    fhir_proxy.post_resource("Observation", file_name, token)
+    logging.info("Posted Observation resource via Proxy.")
+
+    max_wait_seconds = 60
+    poll_interval = 5
+    waited = 0
+    while waited < max_wait_seconds:
+        current_audit_event_count = hapi.get_audit_event_count()
+        if current_audit_event_count == initial_audit_event_count + 1:
+            logging.info("AuditEvent created successfully after POST.")
+            return
+        time.sleep(poll_interval)
+        waited += poll_interval
+        logging.info("Waiting for AuditEvent... (%ds elapsed)", waited)
+
+    raise AssertionError(
+        "AuditEvent was not created after POST within {} seconds." 
+        "Initial count: {}, Current count: {}".format(max_wait_seconds, initial_audit_event_count, current_audit_event_count)
+    )
+
+@patch.dict(os.environ, {"AUDIT_EVENT_LOGGING_ENABLED_ENV": "true"})
+def test_post_bundle_with_logging_enabled_creates_audit_events(
```

Comment:
This and the previous function are very similar, please refactor the common part.